src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -146,11 +146,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         }
                     }
                 } else {
-                    foreach ($data as $value) {
-                        if ($choiceList->getChoicesForValues([$value])) {
-                            $knownValues[] = $value;
-                            unset($unknownValues[$value]);
-                        }
+                    foreach (array_keys($choiceList->getChoicesForValues($data)) as $index) {
```

Comment:
no need to use `array_keys` here, which would iterate twice on the array (as `array_keys` has to iterate). Use `foreach ($choiceList->getChoicesForValues($data) as $index => $choice)` instead

Reply:
Fixed. (see https://github.com/symfony/symfony/pull/48342)